# Исследование надежности заемщиков


## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()
    

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum().to_frame()

0
children          0
days_employed     0
dob_years         0
education         0
education_id      0
family_status     0
family_status_id  0
gender            0
income_type       0
debt              0
total_income      0
purpose           0

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

In [17]:
data.duplicated().sum()

0

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [18]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [19]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [20]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [21]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [22]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Для изучения вляния  профиля заемщика на своевременность платежей воспользуемся метрикой -
**долей клиентов с задолженностью**,  **далее метрика**,  которую определим как отношение числа клиентов, имевших     
задолженность к общему числу клиентов. Вычислим и проанализируем эту метрику.  

проанализируем столбец `debt`:

In [23]:
data['debt'].value_counts()

0    19599
1     1732
Name: debt, dtype: int64

Вероятно значение 0 присвоено клиентам без задолженности. Cгруппируем таблицу по количеству детей и 
вычислим долю клиентов с задолженностью.
Результат сохраним в таблице `children_data_grouped`, где
`children`   - количество детей;
`sum`        - клиенты, имевшие задолженность;
`count`      - число клиентов;
`mean`       - доля клиентов с задолженностью;
`total_mean` - общая доля всех клиентов с задолженностью.





In [24]:
children_data_grouped = data.groupby('children').agg({'debt':['sum','count','mean']})
children_data_grouped ['total_mean'] = data['debt'].mean().round(3)
children_data_grouped [('debt','mean')] = children_data_grouped [('debt','mean')].round(3)
children_data_grouped 

debt               total_mean
           sum  count   mean           
children                               
0         1063  14091  0.075      0.081
1          444   4808  0.092      0.081
2          194   2052  0.095      0.081
3           27    330  0.082      0.081
4            4     41  0.098      0.081
5            0      9  0.000      0.081

<div class="alert alert-success">
<b>Комментарий ревьюера №1</b> ✅
    
Могу посоветовать переименовать столбцы и написать функцию для построения сводных во всех вопросах.
</div>

**Вывод:** Анализ таблицы показывает несущественные различия значений анализируемой метрики у
клиентов с  детьми: разница между клиентами с  одним, двумя и четырьмя детьми составляет около 1%. Однако данные о клиентах с более чем двумя детьми недостаточны для статистически значимых выводов из-за низкого числа клиентов в данной категории.
В тоже время метрика этой категории клиентов не ниже, чем метрика по всей базе.
 Количество детей существенно не влияет на анализируемую  метрику.
Для клиентов без детей значение метрики значимо ниже, как в сравнении с  клиентами, имеющих детей, так  и для всех клиентов в целом. 

**Клиенты без детей имеют меньше просроченных кредитов, чем клиенты с детьми, а также меньше чем клиенты по всей базе в целом.
Однако количество детей оказывает незначительное влияние на возврат кредита в срок.
Доля анализа клиентов с более чем двумя детьми, недостаточно статистических данных.** 

           

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Сгруппируем таблицу по столбцу `family_status`и отсортируем данные по возрастанию значения метрики

In [25]:
# Ваш код будет здесь. Вы можете создавать новdata_grouped = data.groupby('children')\.agg({'debt':['sum','count','mean']})
famaly_data_grouped = data.groupby('family_status').agg({'debt':['sum','count','mean']})
famaly_data_grouped['total_mean'] = data['debt'].mean().round(3)
famaly_data_grouped[('debt','mean')] = famaly_data_grouped[('debt','mean')].round(3)
famaly_data_grouped.sort_values(by=('debt','mean'))

debt               total_mean
                       sum  count   mean           
family_status                                      
вдовец / вдова          63    951  0.066      0.081
в разводе               84   1189  0.071      0.081
женат / замужем        927  12261  0.076      0.081
гражданский брак       385   4134  0.093      0.081
Не женат / не замужем  273   2796  0.098      0.081

**Вывод:** Семейное положение оказывает существеннное значение на показание метрики.
Наиболее надежные заемщики клиенты в категорях "вдова/вдовец" и "в разводе", значение метрики в этих
категориях  значительно ниже, чем у клиентов в гражданском браке или не состоявших в официальном браке. Клиенты, не состоявшие в официальном браке, относятся к категории с повышенным риском
**Клиенты категорий: "вдовец/вдова", "в разводе", "женат/замужем" имеют значение метрики ниже среднего значения по всей выборке в целом. Клиенты из категории "Не женат/замужем" имеют высокий риск задолженности.**

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Сгруппируем столбец `total_income_category`, значения в котором соответствуют следующим интервалам дохода:

**0–30000 — E;
 30001–50000 — D;
 50001–200000 — C;
 200001–1000000 — B;
 1000001 и выше — A.** 
 и отсортируем по возрастанию  исследуемой метрики:

In [26]:
income_data_grouped = data.groupby('total_income_category').agg({'debt':['sum','count','mean']})
income_data_grouped['total_mean'] = data['debt'].mean().round(3)
income_data_grouped[('debt','mean')] = income_data_grouped[('debt','mean')].round(3)
income_data_grouped.sort_values(by=('debt','mean'))

debt               total_mean
                        sum  count   mean           
total_income_category                               
D                        21    349  0.060      0.081
B                       354   5014  0.071      0.081
A                         2     25  0.080      0.081
C                      1353  15921  0.085      0.081
E                         2     22  0.091      0.081

В предложенных  категориях уровня дохода клиенты распределены неравномерно, так в категориях, A, E
число клиентов мене 0.1%, в категории D менее 2%  от общего числа клиентов. Тем не менее, сравнивая
данные столбцов B и C, размер которых достаточен для сравнения, сделаем предположение :  
**c увеличением уровня доходов, риск невозврата кредита в срок уменьшается**.

Для его проверки произведем категоризацию столбца `total_income`придерживаясь следующей методики выбора интервалов: найдем медиану столбца,  затем полученные два списка  разделим медианными значениями, создадим новый столбец `total_income_category_two`cо следующими значениями:
`ниже среднего`.`средний`,
`выше среднего`,`высокий`


In [27]:
# создадим список, включающий значения медиан и вычислим их отсортируем по возрастанию
income_median = [(data['total_income'].median())]
income_median.append(data.loc[data['total_income'] >= income_median[0] ,'total_income'].median())
income_median.append(data.loc[data['total_income'] < income_median[0] ,'total_income'].median())
income_median.sort()
income_median

[103423.0, 142594.0, 188755.5]

In [28]:
# функция, возвращающая значение категории в зависимости от интервала между значениями в income_median
def cotegorize_total_median (income):
    try:
        if income <= income_median[0]:
            return 'ниже среднего'
        elif income_median[0] < income <=income_median[1]:
            return 'средний'
        elif income_median[1] < income <=income_median[2]:
            return 'выше среднего'
        elif  income >= income_median[2]:
            return 'высокий'
    except:
        return 'нет категории'
# создание total_income_category_two и присвоение ему значений.        
data['total_income_category_two'] = data['total_income'].apply(cotegorize_total_median )    
data.head()

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose total_income_category          purpose_category  \
0               покупка жилья                     B  операции с недвижимостью   
1     приобретение автомобиля                     C    операции с автомобилем   
2               покупка жилья                     C  операции с недвижимостью   
3  дополнительное образование                     B     получение образования   
4             сыграть свадьбу                     C        проведение свадьбы   

  total_income_category_two  
0                   высокий  
1                   средний  
2             выше среднего  
3                   высокий  
4             выше среднего

Сгруппирум столбец total_income_category и отсортируем по возрастанию исследуемой метрики:

In [29]:
income_data_grouped = data.groupby('total_income_category_two').agg({'debt':['sum','count','mean']})
income_data_grouped['total_mean'] = data['debt'].mean().round(3)
income_data_grouped[('debt','mean')] = income_data_grouped[('debt','mean')].round(3)
income_data_grouped.sort_values(by=('debt','mean'))

debt              total_mean
                           sum count   mean           
total_income_category_two                             
высокий                    428  5807  0.074      0.081
ниже среднего              384  4859  0.079      0.081
выше среднего              397  4741  0.084      0.081
средний                    523  5924  0.088      0.081

**Вывод:** анализ результатов в таблице позволяет сделать следующие выводы:
клиенты с наиболее высоким доходом имеют самую низкую вероятность невозврата кредита. Наиболее высокую вероятность невозврата кредита имеют клиенты  со средним уровнем дохода. У клиентов с уровнем дохода ниже среднего показатель метрики ниже, чем в целом по базе. У тех, чей доход выше среднего, значение метрики незначительно превышает общий показатель. В целом существует зависимость между уровнем дохода и возвратом кредита в срок. Характер этой зависимости не выявлен полностью в данном исследовании. **Предположение о том, что c увеличением уровня доходов, риск невозврата кредита в срок уменьшается подтверждено только для клиентов с высоким уровнем дохода.**

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

Для проведения анализа сгруппируем таблицу по значениям из столбца`purpose_category`:

In [30]:

income_data_grouped = data.groupby('purpose_category').agg({'debt':['sum','count','mean']})
income_data_grouped['total_mean'] = data['debt'].mean().round(3)
income_data_grouped[('debt','mean')] = income_data_grouped[('debt','mean')].round(3)
income_data_grouped.sort_values(by=('debt','mean'))

debt               total_mean
                          sum  count   mean           
purpose_category                                      
операции с недвижимостью  780  10751  0.073      0.081
проведение свадьбы        183   2313  0.079      0.081
операции с автомобилем    400   4279  0.093      0.081
получение образования     369   3988  0.093      0.081

**Вывод:** Цель кредита оказывает значительное влияние на показатель метрики. Самыми надежными заемщиками оказались клиенты, берущие кредит на "операции с недвижимостью", у которых значение метрики составляет 0.073. Это значение ниже, чем у клиентов, взявших кредит на образование и покупку автомобиля (0.093).Также надежными являются клиенты, взявшие кредит на проведение свадьбы, для них показатель составляет 0.079. 

**Наиболее рискованными оказались клиенты, берущие кредит на "операции с автомобилем" и "образование", как по сравнению с предыдущими категориями, так и по отношению ко всем клиентам в целом.**






#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Причин возникновения пропусков может быть несколько: отсутствие информации или желание её скрыть, технические проблемы, такие как ошибки, приведшие к утере данных и неприменимость некоторых атрибутов к конкретному объекту. Например, атрибут 'женат/замужем' неприменим к ребенку, что также может привести к появлению пропусков в данных.


#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* в количественных переменных могут встречаться значения, которые явно выделяются из общего ряда. Среднее арифметическое такого ряда может значительно отличаться от логически ожидаемого значения и, соответственно, привести к неточностям в анализе. Для исключения такой ситуации применяют медианное значение. Суть его заключается в упорядочивании числового ряда по значениям и нахождении значения, соответствующего среднему расположению в ряду. Если отсутствуют выбросы или аномальные значения, можно использовать среднее арифметическое значение.


### Шаг 4: общий вывод.

В проекте проведено исследование влияния атрибутов профиля заемщика: количества детей, семейного положения, уровня дохода и целевого использования средств на значение метрики - **доля клиентов с задолженностью**.Также в процессе предобработки данных были удалены пропуски, обработаны дубликаты, произведена категоризация и изменены типы данных. Результаты исследования позволяют сделать вывод о влиянии исследуемых атрибутов на целевую метрику.

Наибольшее влияние оказало семейное положение: наиболее рискованной группой являются клиенты в котегории "не женат/не замужем" с показателем 9.8% **Это самое высокое значение метрики в исследовании** , в то время как наименьший процент задолженности у клиентов в категории "вдовец/вдова" – 6.6% - **самое низкое значение в исследовнии**.

Значимое влияние оказывает целевое назначение кредита: наибольший процент задолженности у клиентов, берущих кредит на образование и покупку автомобиля со значением метрики, равным 9.3%.
Клиенты, берущие кредит на покупку недвижимости, представляют группу наиболее надежных заемщиков, для них значение метрики составило 7.3%.

Уровень дохода оказывает менее существенное влияние на метрику: у клиентов с высоким уровнем 
дохода - 7.4%, а у клиентов с доходами ниже среднего, выше среднего и среднего  – 7.9%, 8.4% и 8.8% соответственно.

Одним  факторов, влияющих на кредитоспособность, является наличие детей. У клиентов без детей значение метрики составляет 7.5%, а у клиентов с одним или двумя детьми – 9.2% и 9.5% соответственно. Данные о клиентах с более чем двумя детьми представлены недостаточно.

Рекомендуется произвести аналогичные исследования по другим атрибутам профиля заемщика, а также
рассмотреть целесобразность  разработки  модели прогнозирования целевой метрики в зависимости от атрибутов профиля заемщики на основе методов машинного обучения.